### PR TITLE
Allow to use environment variable to change configuration settings

### PR DESCRIPTION
**Pull request by kouz75**
We use bolt on several environment (dev/test/production).
On each environment, we need to change some settings (DB,debug,API key etc).
using _local.yml allow to do the job but the files aren't save on git and we cannot share them. 

Our solution is to create a environnement variable to define the current plateform and use it to  read  setting files for the current environnement.

### CODE REVIEW

File: src/Config.php [hunk 1]
```diff
@@ -322,6 +322,12 @@ protected function parseGeneral()
         // Read the config and merge it. (note: We use temp variables to prevent
         // "Only variables should be passed by reference")
         $tempconfig = $this->parseConfigYaml('config.yml');
+        if(isset($_SERVER['ENVIRONMENT']))
```
Comment (GawainLynch) [inline src/Config.php]: Super globals should never be used

File: src/Config.php [hunk 2]
```diff
@@ -322,6 +322,12 @@ protected function parseGeneral()
...
+        {
+            $tempconfigenvironment = $this->parseConfigYaml('config_'.$_SERVER['ENVIRONMENT'].'.yml');
```
Comment (GawainLynch) [inline src/Config.php]: PSR-2 on the variable names, e.g. `$tempConfigEnvironment`

File: src/Config.php (hunk 1 above)
Comment (GawainLynch) [inline src/Config.php]: The other problem I have with this approach is that it is putting config back into the request cycle to finish … @CarsonF any ideas on approach here?

**Comment by GawainLynch** [comment]
As this is feature based, I am moving it to `release/3.4` as 3.2 is hotfix only, and 3.3 is in beta freeze.

**Comment by rossriley** [comment]
This is a similar feature to #6208, not sure whether we need both approaches. 

If we do include this method too, then I'd also suggest that we use `getenv('environment')` rather than accessing the superglobal directly.

File: src/Config.php (hunk 1 above)
Comment (kouz75) [inline src/Config.php]: fix super globals

File: src/Config.php (hunk 2 above)
Comment (kouz75) [inline src/Config.php]: I have rewrite several variable name

**Comment by CarsonF** [comment]
I think this is different enough from your other PR @rossriley. 

We at my work do something similar. We have sections in one config file and each environment can extend another so we don't have to duplicate common values. We use `PHP_ENV` though for our env key.

While I do think config_local.yml isn't good enough, it seems to work fine for everyone else. I'm wondering if this isn't something that should be customized for your own install. I'm pretty sure that logic will be more accessible after @gawainlynch's config refactor drops.

**Comment by GawainLynch** [comment]
This was discussed at tonight's core [development meeting](https://github.com/bolt/bolt/wiki/Dev-meeting-2017-01-24) and we feel the fit as-is isn't correct. As noted above and in the meeting, similar functionality already exists, and a whole lot more will be possible after the config loading refactor lands in 3.4 next month, so this would only lock in BC requirements that are going to change very soon.

All that aside, thank you very much for the PR, all-in-all the idea is great, but the base of Bolt needs work to better do this in a supportable way :+1:

**Pull request closed without merging**